### ISSUE TITLE
ADJs with objects

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/UD_English-EWT#203]
- [EWT](http://match.grew.fr/?corpus=UD_English-EWT@dev&custom=61143c75d25bb&clustering=N.lemma)
- [GUM](http://match.grew.fr/?corpus=UD_English-GUM@dev&custom=61143d3f0cb01&clustering=N.lemma)

Many of these are misanalyzed relative clauses.

Interesting cases: "worth" ("worth every penny"), "aged" ("aged 21 (years)"). I don't know what the right analysis is.

**Comment by amir-zeldes** [comment]
"like" and "proclaimed" are just errors. For "worth" I actually think obj is not wrong, what else should we call it? In other languages with overt case marking, you actually see accusatives there, Heb. שווה את זה, German "den Preis wert". The German example is even a cognate of worth...

I'm less sure about aged, would you consider it an obj for a verb? Because if you age something, it would be the second object (they aged whiskey 21 years), and in that case I guess whiskey would be obj and the years would have to be tmod, right?

**Comment by nschneid** [comment]
The [`obj` guidelines](https://universaldependencies.org/u/dep/obj.html) speak only of verbs as having objects. "Worth $10" is a special construction...what about `obl:npmod`? Cf. "8 miles long".

"Aged 21" is a different sense than the causative, I think.

**Comment by amir-zeldes** [comment]
I think if a word takes a dependent with overt accusative marking it should really be `obj`... Forcing that to only apply to VERB heads seems very euro-centric. 

For example, in the medium and bunsetsu analyses of Japanese tokenization, desiderative forms of verbs are actually deverbal adjectives which can take objects (so "want to eat a sandwich" can essentially be an adjective with an accusative object, something like being "eating-covetous the sandwich ACC"). I think those things should be allowed to take `obj` (it's really just part of a verb's regular paradigm, and should supposedly take Mood=Des, though I see that's actually not the case in UD Japanese).

And for 'worth", what else would we want the relation to be? An `obl:npmod` relation would suggest an extent adjunct interpretation to me, but that should be omittable. I don't think we can say "???it's worth".

**Comment by manning** [comment]
There has been lots of discussion of _worth_ in the linguistics literature, as an instance of a **transitive adjective**. For an informal intro, you could start with [Mark Liberman](https://languagelog.ldc.upenn.edu/nll/?p=3096) but there are also academic articles on the subject. It doesn't offend me to regard it exceptionally allowing an `obj`.

**Comment by nschneid** [comment]
A lot of the EWT `obj` errors are relative pronouns. What to do with a relative clause on a predicate, e.g.

- If the baby is feathered yet - which I'm sure he is mostly

Right now "which I'm sure he is mostly" is attached as parataxis. But should it be treated as a relative clause? If so, should "which" be the head of the RC as it stands for the predicate, and should there be `ref(feathered, which)` in the edeprels?

**Comment by nschneid** [comment]
Another question: What to do about

- say what color your chicks are likely to be

I changed this to a free RC analysis along the lines of

https://github.com/jl908069/UD_English-EWT/blob/768cb4926cbc3bf34de2366eff3db034d1b290d5/not-to-release/sources/answers/20111108110044AA4rs9f_ans.xml.conllu#L64-L72

but the wrinkle is that the final "be" is a stranded copula. Should this be treated like preposition stranding (#87), where the copula nonprojectively modifies the head of the RC (along with the infinitive marker)? Or should the "be" be treated as a VERB and attached as xcomp?

**Comment by amir-zeldes** [comment]
> if the baby is feathered yet - which I'm sure he is mostly

I think it should be relative clause from verb to subordinate root ("sure"), compare GUM examples:

http://match.grew.fr/?corpus=UD_English-GUM@dev&custom=61508588ab804

For this case, I think the "which" unusually should be mark, so:

acl:relcl(feathered, sure)
mark(sure, which)

Since "he is mostly" already represents the object clause of "sure", so "which" is redundant and just works as a complementizer indicating the relative clause (compare: "if the baby is feathered, of which I'm sure" - in this case "which" would be obl, since the arg is not otherwise realized, but still not head of the relative clause, which is "sure" in both cases)

**Comment by nschneid** [comment]
I don't follow—"he is" is incomplete. Semantically it is "he is feathered". Doesn't "which" stand for "feathered"?

**Comment by amir-zeldes** [comment]
Oh, yes, I see what you mean! I still think "sure" is the head predicate inside the relative, but you're right, sorry, "which" is the head of a ccomp governed by "sure". Then it should be:

acl:relcl(feathered,sure)
ccomp(sure,which)
cop(which, is)
nsubj(which, is)

So "which" is extracted from the complement clause. Does that look right?

**Comment by nschneid** [comment]
Assume you meant nsubj(which, he).

I thought that was a plausible analysis. But then I went to look for other examples and found:

- Unless the tank has a built in overflow box (which I'm certain it doesn't )

Do we really want to say that "which" stands syntactically for "have a built in overflow box"? Or would it be better to treat "which" as `mark` as you suggested, add an ellipsis node for "have", and connect the whole parenthetical as `parataxis`? If "which" were omitted there would be no reason whatsoever to treat it as a relative clause.

**Comment by nschneid** [comment]
Do you think that it's ellipsis or that the "which" stands for the predicate?

**Comment by amir-zeldes** [comment]
I don't think we need to assume ellipsis, but if there is no function that "which" is fulfilling in the subordinate argument structure, then it can be labeled mark (just marking that this is subordinate)

**Comment by nschneid** [comment]
"I'm certain it doesn't" on its own is incomplete and would require an ellipsis analysis.

What I think you're saying is that "which" here can serve as the predicate embedded under "certain", so 

acl:relcl(has, certain)
nsubj(certain, I)
aux(certain, 'm)
ccomp(certain, which)
nsubj(which, it)
aux(which, does)
advmod(which, n't)

And in edeprels: ref(has, which).

This structure is highly nonprojective, but it parallels the (slightly nonprojective) structure

- the man who I'm certain (that) you've met

in treating the wh-word as extracted.

**Comment by amir-zeldes** [comment]
> "I'm certain it doesn't" on its own is incomplete and would require an ellipsis analysis.

No, I think it's complete and can stand on its own - it's anaphoric, but that's not a syntactic/ellipsis issue IMO.

> What I think you're saying is that "which" here can serve as the predicate embedded under "certain"

Yes and no... I think the which is redundant, since "does" is already sufficient for representing the ccomp. So I think within the subordinate clause it doesn't actually fulfil any valency, only indicating subordination; I would have probably thought the full 'reconstruction' if we need one is:

  * *of* which I'm certain that it doesn't

Which makes the 'which' a backreference to the NP "the tank". I think you are reading "does" as an auxiliary to an unexpressed verb (copy of built), whereas I'm reading it as a standalone anaphoric pro-verb, maybe that's why we see it a little differently?

**Comment by nschneid** [comment]
Yeah—I think CGEL p. 101 characterizes this as the *do* of do-support, and thus an auxiliary, rather than the lexical verb pro-form:

![image](https://user-images.githubusercontent.com/985263/135489761-e57a54aa-c165-479b-8268-d95f9f178748.png)

The pro-form is "but I did __do__", meaning *did* is an auxiliary.

**Comment by nschneid** [comment]
British English idiosyncrasies aside, consider:

- He thought I wrote it, but I didn't.
- He thought I wrote it, but I didn't do so.
- *He thought I wrote it, but I didn't so.

This shows that "didn't" isn't acting as a pro-verb, right?

**Comment by amir-zeldes** [comment]
No, I don't think that shows it, it just shows that the contraction "didn't" is auxiliary, but you can do positive:

  * He thought I wrote it, and indeed, I did so

In the "double do" the first has to be aux, but if you have just one it can be the pro-verb by itself, or an auxiliary standing for the emphatic construction. Otherwise there would be no way to use a pro-verb form without being emphatic.

**Comment by amir-zeldes** [comment]
OK, sorry, actually the tank example has the form doesn't, so auxiliary should be right for that one.

**Comment by nschneid** [comment]
@manning, @sylvainkahane - any opinions on this?

- Unless the tank has a built in overflow box (which I'm certain it does/AUX n't )

We've established that it's the do-support use of "do". Question is whether "which" is acting as a subordinator or has an antecedent, and if so, should it serve as the predicate of the embedded clause (`ccomp(certain, which)`, `nsubj(which, it)`, `aux(which, does)`, etc.).

**Comment by nschneid** [comment]
>   * Unless the tank has a built in overflow box (which I'm certain it does/AUX n't )

Another instance of this construction in https://github.com/UniversalDependencies/UD_English-EWT/issues/269#issuecomment-1019039226:

- if we lose Dean (which we will if we don't pay 65k + 10k)

**Comment by nschneid** [comment]
Back to the original question: EWT and GUM have "worth" and "aged" as ADJs taking objects.

This one seems like an error, though: "better" should be a VERB: https://universal.grew.fr/?custom=676b43b4a5ecb

**Comment by amir-zeldes** [comment]
Will fix instance of 'better'. Regarding the other two, I think "worth" takes an object as discussed above. I think for "aged" I'd be OK with `obl:unmarked`, but if you prefer `obj` I can live with that too - we'd just need to decide which is the `iobj` in two object cases (aged the whiskey 25 years). But it feels more like an extent adverbial to me TBH.

**Comment by nschneid** [comment]
Fine with keeping "worth" as is.

I agree that *transitive verb* "age" should have `obl:unmarked`, but the adjective sense we are dealing with here ("He died aged 96.") is distinct. E.g. you can't add *for* to indicate a duration:

- *He died aged for 96.
- whiskey aged (for) 25 years

**Comment by amir-zeldes** [comment]
Hm, I can live with either, though my gut is still obl for aged, which has a clear verbal correspondent (even if "for" doesn't work). But if you prefer obj that's OK by me too.
